pkg/services/accesscontrol/resourcepermissions/store.go

```diff
@@ -739,21 +742,42 @@ type ActionSet struct {
 
 // InMemoryActionSets is an in-memory implementation of the ActionSetService.
 type InMemoryActionSets struct {
-	log        log.Logger
-	actionSets map[string][]string
+	log                log.Logger
+	actionSetToActions map[string][]string
+	actionToActionSets map[string][]string
 }
 
 // NewActionSetService returns a new instance of InMemoryActionSetService.
-func NewActionSetService() ActionSetService {
-	return &InMemoryActionSets{
-		actionSets: make(map[string][]string),
-		log:        log.New("resourcepermissions.actionsets"),
+func NewActionSetService(a *acimpl.AccessControl) ActionSetService {
+	actionSets := &InMemoryActionSets{
+		log:                log.New("resourcepermissions.actionsets"),
+		actionSetToActions: make(map[string][]string),
+		actionToActionSets: make(map[string][]string),
+	}
+	a.RegisterActionResolver(actionSets)
+	return actionSets
+}
+
+func (s *InMemoryActionSets) Resolve(action string) []string {
+	actionSets := s.actionToActionSets[action]
+	sets := make([]string, 0, len(actionSets))
+
+	for _, actionSet := range actionSets {
+		prefix := strings.Split(actionSet, ":")[0]
```

Comment:
nit: should we make sure that the actionSet is splitable here?

want to really avoid runtime errors here